lib/index.js

```diff
@@ -200,13 +199,29 @@ class PWMetrics {
 
   findMedianRun(results) {
     const ttiValues = results.map(r => r.timings.find(timing => timing.id === metrics.ids.TTI).timing);
-    const medianTTI = median(ttiValues);
+    const medianTTI = this.getMedianRun(ttiValues);
     // in the case of duplicate runs having the exact same TTI, we naively pick the first
     const medianRun = results.find(result => result.timings.find(timing => {
       return timing.id === metrics.ids.TTI && timing.timing === medianTTI;
     }));
     return medianRun;
   }
+
+  getMedianRun(values) {
+    if (values.length == 1) {
+      return values[0];
+    }
+
+    values.sort(function sortValues(a, b) {
+      return a - b;
+    });
+
+    const half = Math.floor(values.length / 2);
+    if (values.length % 2)
+      return values[half];
+    else
+      return values[half]; // take slightly worse entry as median.
```

Comment:
It's a mistake, thanks for catching it, just commited a fix. There was also the case where values.length is 2 and the half variable already is correct index.